### ISSUE TITLE
sll handshake error

### ISSUE THREAD

**Issue by sunjunlishi** [googlesamples/assistant-sdk-cpp#7]
assistant_sdk robots_pem:
assistant_sdk CreateCustomChannel(embeddedassistant.googleapis.com:443, creds, arg)

E1125 21:57:19.169789182 3080 ssl_transport_security.c:940] Handshake failed with fatal error SSL_ERROR_SSL: error:14090086:SSL routines:ssl3_get_server_certificate:certificate verify failed.

**Comment by sunjunlishi** [comment]
sudo ./run_assistant --audio_input ./resources/switch_to_channel_5.raw --credentials_file ./credentials.json --credentials_type USER_ACCOUNT
assistant_sdk robots_pem: 
assistant_sdk CreateCustomChannel(embeddedassistant.googleapis.com:443, creds, arg)

E1126 03:29:12.620437994   25929 ssl_transport_security.c:940] Handshake failed with fatal error SSL_ERROR_SSL: error:14090086:SSL routines:ssl3_get_server_certificate:certificate verify failed.
E1126 03:29:14.873731526   25929 ssl_transport_security.c:940] Handshake failed with fatal error SSL_ERROR_SSL: error:14090086:SSL routines:ssl3_get_

**Comment by sunjunlishi** [comment]
https://pki.google.com/roots.pem, i down it,thats works!

**Comment by sunjunlishi** [comment]
stream->WritesDone();Can stream be called two times?
second time error
void Channel::PerformOpsOnCall(CallOpSetInterface* ops, Call* call) {
  static const size_t MAX_OPS = 8;
  size_t nops = 0;
  grpc_op cops[MAX_OPS];
  ops->FillOps(call->call(), cops, &nops);
  GPR_ASSERT(GRPC_CALL_OK ==
             grpc_call_start_batch(call->call(), cops, nops, ops, nullptr));
}

**Comment by Fleker** [comment]
I believe once `stream->WritesDone()` is called then you can no longer write to the stream.

**Comment by sunjunlishi** [comment]
CreateChannel(endpoint)->wrteDone
SSL validation is created every time it is sent, and it takes 2 seconds to create the channel

**Comment by sunjunlishi** [comment]
So when I had to speak, I did the channel creation

**Comment by sunjunlishi** [comment]
ConverseResponse response;
while (stream->Read(&response))
{  // Returns false when no more to read.
			std::cout << "assistant_sdk Got a response \n";

			if ((response.has_error() || response.has_audio_out() ||
				response.event_type() == ConverseResponse_EventType_END_OF_UTTERANCE)
				/*&& audio_input->IsRunning()*/)
			{}
}





PUT it into the thread and crash

**Comment by Fleker** [comment]
I don't know if this is thread-safe.

**Comment by sunjunlishi** [comment]
So it blocks the processing of the voice, does the input and output of the voice have to be placed in the thread

**Comment by sunjunlishi** [comment]
ok,i am put all in one thread

bool gbassistantInitok = true;
bool gbAssisThreadWork = false;
bool gbGetResponse = false;
void DoInitFunc()
{
	if (gbassistantInitok)
	{
		gbassistantInitok = false;
		gbAssisThreadWork = true;
		printf("********** gbassistantInitok false\n");
		gassistant_init();
		printf("你可以呼叫我了******* gbassistantInitok true\n");
		gbassistantInitok = true;
		while (gbAssisThreadWork)
		{
			if (gbGetResponse)
			{
				gassistant_writeDone();
				gassistant_responseok();
				gbAssisThreadWork = false;
				gbGetResponse = false;
			
				return;
			}
			Sleep(100);
		}
	}

}

**Comment by Fleker** [comment]
Does that work?

**Comment by sunjunlishi** [comment]
yes，
 std::thread   t1(DoInitFunc);
t1.detach();


for (;;) 
{
        if ((k = ad_read(ad, adbuf, 2048)) > 0)
       {
                          gassistant_writeFirst();
			 gassistant_writeData((char*)adbufTmp, 2 * k);
       }
      gbGetResponse = true;
}

Initialization and reception do not affect speech processing

**Comment by sunjunlishi** [comment]
I found that Fast response of Google audio,my is slowly very. i  your  google home use the  same code?
std::shared_ptr<ClientReaderWriter<ConverseRequest, ConverseResponse>>
		stream(std::move(assistant->Converse(&context)));

**Comment by sunjunlishi** [comment]
Every time you need to re create the object, to shake hands

**Comment by Fleker** [comment]
The `ConverseRequest` represents a single instance of a request. A new one should be constructed for each request.

The audio playback is faster or slower than normal? It may be a difference in how you have configured your AudioOutConfig. Make sure you set a consistent sample rate.

**Comment by sunjunlishi** [comment]
on it is  not audio playback . 
      google                 ConverseRequest---->ConverseResponse  complete cost 1 second
my   stream init         ConverseRequest---->ConverseResponse  complete cost 6 second

**Comment by Fleker** [comment]
I don't believe the Google Home uses this SDK. I'm unsure of which part is creating the latency. Perhaps it is related to the CPU, or perhaps there's different audio compression being used.

**Comment by sunjunlishi** [comment]
oh，Parsing is very detailed

**Comment by sunjunlishi** [comment]
mpChannel = CreateChannel(api_endpoint);
assistant = std::unique_ptr<EmbeddedAssistant::Stub>(
EmbeddedAssistant::NewStub(mpChannel));

could be use more times

**Comment by sunjunlishi** [comment]
Is it possible that I need to apply for commercial development, the interface voice of the commercial development interface returns faster

**Comment by Fleker** [comment]
Right now we're in a developer preview, so there are no applications for commercial development at the moment.

**Comment by sunjunlishi** [comment]
ok,thanks when will.

**Comment by Fleker** [comment]
I do not have a specific date. We're continuing to develop the SDK so that it will be in a good place for developers.

**Comment by sunjunlishi** [comment]
i found that  the server response more quickly

**Comment by sunjunlishi** [comment]
Under the SVN network on my computer, it's 1.4s;in  Hong Kong, reponse cost 0.9s
